### PR TITLE
added more edge cases for the getRecommendedSlots function

### PR DESCRIPTION
**Pull request by SemanBoon**
What type of PR is this?

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR includes additional unit tests for the getRecommendedSlots function, covering more edge cases to ensure the function's strength. Specifically, it tests the behavior when a user enters a date in the past, service durations with non-standard lengths, and cases where no user preferences match available slots.

## Screenshots, Recordings

Screenshot 2024-07-31 at 12 06 25 AM

## Test Plan

Manual Inspection: By running these tests, we can confirm that our function correctly calculates available slots under various conditions. The process involved manually checking the accuracy of test results and ensuring they covered all possible cases.

### CODE REVIEW

File: Backend/index.js [hunk 1]
```diff
@@ -618,20 +617,25 @@ app.post('/check-favorite', async (req, res) => {
...
-        return res.status(404).json({ error: 'Service provider not found' });
-      }
-      const availableSlots = getAvailableSlots(provider.schedule, serviceDuration, {}); // Slot popularity passed as {}
-      res.status(200).json(availableSlots);
-      } catch (error) {
-      console.error('Error fetching available slots:', error);
-      res.status(500).json({ error: 'Failed to fetch available slots' });
+    const today = new Date();
+
+    if (userDate < today) {
```
Comment (karen-ka) [inline Backend/index.js:624]: nice handling!

File: Backend/__tests__/utilityFunctions.test.js [hunk 2]
```diff
@@ -34,7 +35,154 @@ describe('getAvailableSlots', () => {
...
+        const userTime = new Date('2021-09-01T08:00:00');
+        const serviceDuration = 2;
+        const preferredPeriod = 'morning';
+        const weights = getUpdatedWeights('popular_slots');
+
+        const recommendedSlots = getRecommendedSlots(slots, userPreferences, provider, userTime, serviceDuration, preferredPeriod, weights);
+        expect(recommendedSlots).toEqual([]);
+    });
+
+    it('should calculate higher score for slots closer to user-entered time', () => {
```
Comment (karen-ka) [inline Backend/__tests__/utilityFunctions.test.js:166]: Good test integrating multiple factors and testing the tiebreaker.

File: frontend/src/Components/BookingPage/BookingPage.jsx [hunk 3]
```diff
@@ -69,7 +75,6 @@ const BookingPage = () => {
                             userTime: time,
                         }),
                     });
-                    console.log("priority", userPriority)
```
Comment (karen-ka) [inline frontend/src/Components/BookingPage/BookingPage.jsx:72]: good catch - let's make this the last one of these.

**Review by karen-ka — APPROVED** [review]
